### PR TITLE
feat: sync with `meilisearch` for ruby

### LINKED ISSUE #196: 🚀 Feature: Add Sync with Meilisearch for Ruby
Appwrite Functions allow you to extend and customize your Appwrite server functionality by executing your custom code. 🤩 You can learn more at our official [Appwrite Functions docs](https://appwrite.io/docs/functions).

Your task is to implement the Sync with Meilisearch function in Ruby. You can look at the existing [Appwrite Functions Templates](https://github.com/appwrite/templates) in another coding language.

Your function should behave and be implemented similarly to existing Sync with Meilisearch Node.js template.

### Tasks summary:

- [ ] Fork & clone Appwrite's [templates](https://github.com/appwrite/templates) repository
- [ ] Create a branch `feat-implement-sync-with-meilisearch-ruby`
- [ ] Write code for the function in folder `ruby/sync-with-meilisearch/`. 
- [ ] Test the function using Appwrite, **provide a video**.
- [ ] Write a `README.md` explaining what the function does, how to use it, and customize it. Here is [an example](https://github.com/appwrite/templates/tree/main/node/sync-with-meilisearch).
- [ ] Submit a pull request to our [templates](https://github.com/appwrite/templates) repository

If you need any help, contact us on our [Discord server](https://discord.gg/GSeTUeA).

Are you ready to work on this issue? 🤔 Let us know, and we will assign it to you 😊

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by afzal442**
## What does this PR do?

Fixes #196 

Test Vid:

https://github.com/appwrite/templates/assets/11625672/0bda0fc7-e69b-48c4-bd91-23cadc396e9c

### CODE REVIEW

File: ruby/meilisearch/config.rb
Comment (loks0n) [inline ruby/meilisearch/config.rb:1]: Can we just use ENV['FOO'] directly? We want to keep it consistent with the other templates and I don't think the pattern here provides enough benefit in this case

File: ruby/meilisearch/Gemfile
Comment (loks0n) [inline ruby/meilisearch/Gemfile:1]: Can we have the template in `ruby/sync_with_meilisearch`?

File: ruby/meilisearch/lib/utils.rb
Comment (loks0n) [inline ruby/meilisearch/lib/utils.rb:1]: None of these functions are used!

Can we use them to match the behaviour of the node runtime

File: ruby/meilisearch/lib/main.rb
Comment (loks0n) [inline ruby/meilisearch/lib/main.rb:1]: This file is not is the correct format.

To be a working appwrite function you need to define a `main` function that accepts a context and returns a response using the helper.

See
https://github.com/appwrite/templates/blob/main/ruby/starter/lib/main.rb

**Review by loks0n — CHANGES_REQUESTED** [review]
Thanks for the PR!

I've added some comments for you to address

File: ruby/meilisearch/lib/utils.rb
Comment (afzal442) [reply ruby/meilisearch/lib/utils.rb:1]: sorry @loks0n any chance?

Comment (afzal442) [reply ruby/meilisearch/lib/utils.rb:1]: cc @gewenyu99

Comment (loks0n) [reply ruby/meilisearch/lib/utils.rb:1]: You can join our discord community and get help in our Hacktoberfest channel.

https://appwrite.io/discord

File: ruby/sync_with_meilisearch/lib/utils.rb
Comment (loks0n) [inline ruby/sync_with_meilisearch/lib/utils.rb:1]: Again, none of these functions are used...

Please change the behaviour of main.js to match that of the node function.

For example, throw_if_missing should be used at the start of the function execution to verify the function has the correct ENV variables set.

**Review by loks0n — CHANGES_REQUESTED** [review]
Please also provide a test video showing the function executing on Appwrite

File: ruby/sync_with_meilisearch/lib/main.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  ])
+
+  if context.req.method == 'GET'
+    html = interpolate(get_static_file('index.html'), {
+      'MEILISEARCH_ENDPOINT' => ENV['MEILISEARCH_ENDPOINT'],
+      'MEILISEARCH_INDEX_NAME' => ENV['MEILISEARCH_INDEX_NAME'],
+      'MEILISEARCH_SEARCH_API_KEY' => ENV['MEILISEARCH_SEARCH_API_KEY']
+    })
+
+    context.response.set_body(html)
```
Comment (afzal442) [inline ruby/sync_with_meilisearch/lib/main.rb:23]: Hi @stnguyen90 Any review and suggestions will be review.

Comment (afzal442) [reply ruby/sync_with_meilisearch/lib/main.rb:23]: cc @loks0n

**Review by loks0n — CHANGES_REQUESTED** [review]
Please also provide a test video showing the function executing on Appwrite

**Comment by afzal442** [comment]
Sorry @loks0n I have been getting this below error. Yesterday, there were no issues, working fine. Any help?
```
Error 400
Execution error: Server Error

Type

general_argument_invalid
```

Otherwise, it's working fine.

**Comment by afzal442** [comment]
Hi @loks0n a test video added in the issue desc box. Plz review it.

**Review by loks0n — APPROVED** [review]

**Comment by afzal442** [comment]
my discord id is **afzal442**

cc @tessamero

**Comment by gewenyu99** [comment]
Thanks, be in touch soon!

**Pull request merged**